### DISCUSSION TITLE
How do I deal with BINARY_EVENT data?

### DISCUSSION THREAD

**Discussion by tjad** [zishang520/socket.io#67] [Announcements]
When event comes to server as BINARY_EVENT, the socket.io server does not see any data in the event listener.
```
		client.On("bindata", func(datas ...any) {
			log.Println("Data received:", datas)
		})
```

How do I deal with this?

**Comment by zishang520** [comment]
server:
```golang
import "github.com/zishang520/engine.io-go-parser/types"

client.On("bindata", func(args ...any) {
	utils.Log().Info("%#v", args[0].(types.BufferInterface))
})
```

client:
```javascript
socket.emit("bindata",new Uint8Array([0x10, 0x20, 30, 40]))
```

**Comment by zishang520** [comment]
Binary content has invisible content, so you need to use `%#v` instead of `%v` to debug, log.Println is equivalent to `%v`.

**Comment by tjad** [comment]
Thank you very much @zishang520 for fast response!

This was not clear from the README documentation - perhaps useful to add.

All is working! Thank you again 🙏

**Reply by zishang520** [reply]
Thanks for your suggestions, I'll deal with them when I have some free time.
